File: Backend/src/main/java/com/askend/filters/model/Filter.java
package com.askend.filters.model;

import jakarta.persistence.*;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@ToString
@Entity
@Table(name = "filters")
public class Filter {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    private String name;

    @NotNull
    @OneToMany(mappedBy = "filter", cascade = CascadeType.ALL, fetch = FetchType.EAGER,
            orphanRemoval = true)
    private List<Criteria> criteriaList;
}

File: frontend/src/app/models/filter.ts
export interface Filter {
    id?: number;
    name: string;
    criteriaList: Criteria[];
}

export interface Criteria {
    id?: number;
    type: 'AMOUNT' | 'TITLE' | 'DATE';
    criteria_condition: string;
    criteria_value: string;
    
}
